Create RainbowDAO-Protocol-ink!-Phase1-milestone-1.md

RainbowcityKun opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [ ] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [ ] This pull request is being made by the same account as the accepted application.
- [ ] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [ ] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/XXX < please fill this in with the PR number of your application.

RainbowcityKun commented:
Hi, friend of the web3 foundation. In the original application I submitted, there were spaces in the grant name, which caused the current inspection to fail. How do I handle this situation now?

https://github.com/RainbowcityFoundation/Grants-Program/blob/7393e564d87ad2b87e87bae1b584f15d0c5fdc6b/applications/RainbowDAO%20Protocol%20ink!%20%20Phase%201.md

Noc2 commented:
Thanks for the delivery. We will look into it as soon as possible.The github checks are mostly for us, so don’t worry about it.

RainbowcityKun commented:
> Thanks for the delivery. We will look into it as soon as possible.The github checks are mostly for us, so don’t worry about it.
OK, thank you very much for your reminder. Feel free to contact me if you have any questions.

RainbowcityKun commented:
> Thanks for the delivery. We will look into it as soon as possible.The github checks are mostly for us, so don’t worry about it.

OK, thank you very much for your reminder. Feel free to contact me if you have any questions.

takahser commented:
@RainbowcityKun thanks again for the delivery.
After taking a look at it, these are some things I noticed:

- The GovnanceDAO tests currently fail
- The docker setup instructions didn't work on my macbook. Also running `docker --mirror Aliyun` fails with: `unknown flag: --mirror`
- I was able to run the frontend code but when testing the protocol functions, all transactions I tried to send failed when using the default settings. Also the instructions are missing on how I can use the UI using a local node.

Could you take a look at these or comment on it?

RainbowcityKun commented:
Hi @takahser, thank you so much for your hard work, we've updated a few of your questions, please recheck to see if it works.

* The GovnanceDAO contract file has been updated and you can download it again.

https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-milestone_1/tree/main/govnance_dao

*  Docker re-updated the documentation, you can see if it works.

  https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-UI-milestone_1/blob/main/docs/docker.md

* The front-end documentation has also been updated, and the links are as follows:

https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-UI-milestone_1#readme

* We have made a more detailed tutorial, the link is as follows:
https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-docs/blob/main/rainbowDaoProtocolTest.md

Feel free to leave us a comment if you have any questions, thank you again for your hard work.

takahser commented:
Hi @RainbowcityKun 

Thx again for the updates!

The following works fine now:
- [x] GovnanceDAO tests

I still have some comments for:
- [ ] Docker docs: instructions on how to start a container from the built image are still missing
- [ ] Rainbow Protocol Tutorial: contrary to the docs, the `build.sh` [script](https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-docs/blob/main/rainbowDaoProtocolTest.md#compile-contracts-from-source-code) doesn't exists

I'll take a look at the updated frontend docs again after the you add the `build.sh` script.

RainbowcityKun commented:
Hi, @takahser ,thank you for your hard work. Now that we've updated these two files again, take a look again.

https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-UI-milestone_1/blob/main/docs/docker.md

https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-milestone_1/blob/main/build.sh

If you find any new problems, feel free to contact me and we will improve them as soon as possible. Thanks again for your hard work!

RainbowcityKun commented:
> And just to add one thing, which I just noticed. The smart contract currently don’t seem to have any [inline documentation](https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-milestone_1/blob/main/erc20_factory/lib.rs). See the contract: [“We will provide both inline documentation of the code”](https://github.com/w3f/Grants-Program/blob/master/applications/RainbowDAO%20Protocol%20ink!%20%20Phase%201.md#milestone-1)

Hi,@Noc2,thank you very much for your careful inspection. We have now updated all contracts and added inline documentation. Please check again and confirm. If you have any questions, leave me a message at any time, and we will update it as soon as possible.

takahser reviewed (changes requested):
Hey @RainbowcityKun 
Thanks for the updates, I'm now fine with the following:

- [x] inline comments
- [x] docker docs
- [x] build script was added

However, I realised that although you added the instructions for [installing a substrate contracts node](https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-docs/blob/main/rainbowDaoProtocolTest.md#1-installing-the-substrate-contracts-node) the following instructions are missing:

- [ ] starting the node
- [ ] connect the local node in the https://polkadot.js.org/apps/
- [ ] the detailed steps on how to get to the interface for uploading the contract (_developer > contracts > Upload & deploy code_ I guess)

I'll go ahead and deploy and test the protocols now before I'll be able to give additional feedback. But feel free to improve the already mentioned points in the meanwhile. :)

RainbowcityKun commented:
> 嘿@RainbowcityKun 感谢您的更新，我现在对以下内容很好：
> 
> * [x]  内联评论
> * [x]  码头工人文档
> * [x]  添加了构建脚本
> 
> 但是，我意识到尽管您添加了[安装基板合同节点](https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-docs/blob/main/rainbowDaoProtocolTest.md#1-installing-the-substrate-contracts-node)的说明，但缺少以下说明：
> 
> * [ ]  启动节点
> * [ ] 在[https://polkadot.js.org/apps/中连接本地节点](https://polkadot.js.org/apps/)
> * [ ] 关于如何进入上传合同界面的详细步骤（我猜_是开发人员>合同>上传和部署代码）_
> 
> 在我能够提供额外反馈之前，我将继续部署和测试协议。但与此同时，请随时改进已经提到的要点。:)

Ok, thank you very much for your careful inspection, and let's keep improving our documentation. We will continue to update, if you have any new questions, feel free to leave us a message.

RainbowcityKun commented:
@takahser The tutorial files are further updated, please check again, thank you again for your hard work.

https://github.com/RainbowcityFoundation/RainbowDAO-Protocol-Ink-docs/blob/main/rainbowDaoProtocolTest.md#1-installing-the-substrate-contracts-node

RainbowcityKun commented:
@takahser , @mmagician ,
Thanks to @takahser for the careful evaluation of our first milestone submission, it also made us realize that there are a lot of small issues, we will double check based on the content of the evaluation report. When committing to milestone 2, we'll tackle all the minor issues together.

Noc2 commented:
Thanks! I’m happy to confirm that the milestone is approved and I will forward your invoice internally.

RainbowcityKun commented:
> Thanks! I’m happy to confirm that the milestone is approved and I will forward your invoice internally.

Thank you very much for your hard work, we will re-optimize our code. The 2nd milestone will also be submitted as soon as possible. We will strive to build the best DAO infrastructure product.

RouvenP commented:
hi @RainbowcityKun we transferred the payment today.
